app/hook/storage.ts

```diff
@@ -5,96 +5,65 @@ import { useStorage } from "~app/storage"
 export { useStorage } from "~app/storage"
 
 export const useAction = () => {
-  return useStorage(
-    useShallow(({ state, ...action }) => {
-      return action
-    })
-  )
+  return useStorage(({ state, ...action }) => {
+    return action
+  })
 }
 
 export const useNetwork = (id?: string) => {
-  return useStorage(
-    useShallow(({ state }) => {
-      const networkId = id ?? state.networkActive
-      return {
-        id: networkId,
-        ...state.network[networkId]
-      }
-    })
-  )
+  return useStorage(({ state }) => {
+    return state.network[id ?? state.networkActive]
+  })
 }
 
 export const useNetworks = () => {
-  return useStorage(
-    useShallow(({ state }) => {
-      return Object.entries(state.network).map(([id, n]) => ({
-        id,
-        ...n
-      }))
-    })
-  )
+  return useStorage(({ state }) => {
+    return Object.values(state.network)
+  })
 }
 
-export const useShouldOnboard = () => {
-  return useStorage(
-    useShallow(({ state }) => {
-      const network = useNetwork()
-      return (
-        Object.values(state.account).filter(
-          (a) => a.chainId === network.chainId
-        ).length === 0
-      )
-    })
-  )
+export const useAccount = (id?: string) => {
+  const network = useNetwork()
+  return useStorage(({ state }) => {
+    return state.account[id ?? network.accountActive]
+  })
 }
 
-export const useAccount = (id?: string) => {
+export const useAccountCount = (networkId?: string) => {
```

Comment:
How about we check how many accounts in `useAccounts` directly? 
It seems that both of them are checking total accounts, and the diff is that `useAccountCount` can return the number of accounts.

Reply:
`useAccounts` would re-render more frequently. Given now we don't encounter performance issue, I think it is a fine choice 👍

Reply:
I see. Thanks for the explanation!